utils/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -25,6 +25,86 @@ contract PreSwapChecker {
   bytes4 constant internal ERC721_INTERFACE_ID = 0x80ac58cd;
   bytes4 constant internal ERC20_INTERFACE_ID = 0x36372b07;
 
+  IWETH public wethContract;
+
+
+  /**
+    * @notice Contract Constructor
+    * @param preSwapCheckerWethContract address
+    */
+  constructor(
+    address preSwapCheckerWethContract
+  ) public {
+    wethContract = IWETH(preSwapCheckerWethContract);
+  }
+
+  /**
+    * @notice If order is going through wrapper to swap
+    * @param order Types.Order
+    * @param fromAddress address
+    * @param wrapper address
+    * @return uint256 errorCount if any
+    * @return bytes32[] memory array of error messages
+    */
+  function checkSwapWrapper(
+    Types.Order calldata order,
+    address fromAddress,
+    address payable wrapper
```

Comment:
why is this payable...?

Reply:
since the wrapper contract can take in ether as a payment in the wrap function.

Reply:
was able to remove wrapper!